lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2442,16 +2447,25 @@ void AttributeChecker::visitImplementsAttr(ImplementsAttr *attr) {
 }
 
 void AttributeChecker::visitFrozenAttr(FrozenAttr *attr) {
-  auto *ED = cast<EnumDecl>(D);
+  if (auto *ED = dyn_cast<EnumDecl>(D)) {
+    if (!ED->getModuleContext()->isResilient()) {
+      diagnoseAndRemoveAttr(attr, diag::enum_frozen_nonresilient, attr);
```

Comment:
This check probably applies to structs as well.

Reply:
Still not resolved.

Reply:
Similar thing, this is the existing behavior, so I'd suggest landing and then a follow-up PR can make things more consistent across the two.